content/blog/2016-09-01-interview-with-shuttlecloud.md

```diff
@@ -0,0 +1,114 @@
+
+![ShuttleCloud's integration with Gmail](/assets/blog/2016-09-07/gmail-integration.png)
+***ShuttleCloud’s [integration](https://support.google.com/mail/answer/164640?hl=en) with Google’s Gmail Platform.*** *Gmail has imported data for 3 million users with our API.*
+
+
+ShuttleCloud’s technology, encrypts all the data required to process an import, in addition to following the most secure standards (SSL, oAuth) to ensure the confidentiality and integrity of API requests. Our technology allows us to guarantee our platform’s high availability, with up to 99.5% uptime assurances. 
+
+![ShuttleCloud by Numbers](/assets/blog/2016-09-07/shuttlecloud-numbers.png)
+
+## What was your pre-Prometheus monitoring experience?
+
+In the beginning, a proper monitoring system for our infrastructure was not one of our main priorities. We didn’t have as many projects and instances as we currently have, so we worked with other simple systems to alert us if anything was not working properly and get it under control.
+
+   * We had a set of automatic scripts to monitor most of the operation metrics for the machines. These were cron-based and executed, using Ansible from a centralized machine. The alerts were emails sent directly to the entire development team.
+   * We trusted Pingdom for external blackbox monitoring and checking that all our frontends were up. They provided an easy interface and alert system in case any of our external services were not reachable.
+
+Fortunately, big customers arrived, and the SLAs started to be more demanding. Therefore, we needed something else to measure how we were performing and to ensure that we were complying with all SLAs. One of the features we required was to have accurate stats about our performance and business metrics (i.e., how many migrations finished correctly), so reporting was more on our minds than monitoring. 
+
+We developed the following system:
+
+![Initial Shuttlecloud System](/assets/blog/2016-09-07/Prometheus-System-1.jpg)
+
+   * The source of all necessary data is a status database in a CouchDB. There, each document represents one status of an operation. This information is processed by the Status Importer and stored in a relational manner in a MySQL database.
+
+   * A component gathers data from that database, with the information aggregated and post-processed into several views. 
+      * One of the views is the email report, which we needed for reporting purposes. This is sent via email. 
+      * The other view pushes data to a dashboard, where it can be easily controlled. The dashboard service we used was external. We trusted Ducksboard, not only because the dashboards were easy to set up and looked beautiful, but also because they provided automatic alerts if a threshold was reached.
+
+With all that in place, it didn’t take us long to realize that we would need a proper metrics, monitoring, and alerting system as the number of projects started to increase. 
```

Comment:
It's "alerting system" here, but then "alert system" multiple times further down. I suggest to call it "alerting system" consistently.